apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -308,19 +309,67 @@ impl ApplyToInternal for NamedSelection {
                     ));
                 }
             }
+
             Self::Group(alias, sub_selection) => {
                 let (value_opt, apply_errors) = sub_selection.apply_to_path(data, vars, input_path);
                 errors.extend(apply_errors);
                 if let Some(value) = value_opt {
                     output.insert(alias.name(), value);
                 }
             }
+
+            Self::Spread(spread) => {
+                let (spread_opt, spread_errors) = spread.apply_to_path(data, vars, input_path);
+                errors.extend(spread_errors);
+                if let Some(JSON::Object(spread)) = spread_opt {
+                    // TODO Better merge strategy for conflicting fields
+                    output.extend(spread);
+                }
+            }
```

Comment:
Calling out this `TODO` as a topic we've previously discussed, but that becomes more urgent with conditional fragments: how should we merge fields (potentially coming from different fragments, with different subselections) when they collide?